File: GXpert/GXpert.Web/Modules/ServerTypes/Playlist/PlayListcontentColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PlayListcontentRow } from "./PlayListcontentRow";

export interface PlayListcontentColumns {
    Id: Column<PlayListcontentRow>;
    PlayListTitle: Column<PlayListcontentRow>;
    ContentTitle: Column<PlayListcontentRow>;
    ExamTitle: Column<PlayListcontentRow>;
    LiveSessionMeetingId: Column<PlayListcontentRow>;
    AssignmentTitle: Column<PlayListcontentRow>;
    ModuleTitle: Column<PlayListcontentRow>;
    SortOrder: Column<PlayListcontentRow>;
    EPublishStatus: Column<PlayListcontentRow>;
    InsertDate: Column<PlayListcontentRow>;
    InsertUserId: Column<PlayListcontentRow>;
    UpdateDate: Column<PlayListcontentRow>;
    UpdateUserId: Column<PlayListcontentRow>;
    IsActive: Column<PlayListcontentRow>;
}

export class PlayListcontentColumns extends ColumnsBase<PlayListcontentRow> {
    static readonly columnsKey = 'Playlist.PlayListcontent';
    static readonly Fields = fieldsProxy<PlayListcontentColumns>();
}

File: GXpert/GXpert.Web/Modules/Playlist/PlayListcontent/PlayListcontentColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Playlist.Columns;

[ColumnsScript("Playlist.PlayListContent")]
[BasedOnRow(typeof(PlayListContentRow), CheckNames = true)]
public class PlayListContentColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string PlayListTitle { get; set; }
    public string ContentTitle { get; set; }
    public string ExamTitle { get; set; }
    public string LiveSessionMeetingId { get; set; }
    public string AssignmentTitle { get; set; }
    public string ModuleTitle { get; set; }
    public float SortOrder { get; set; }
    public short EPublishStatus { get; set; }
}